### ISSUE TITLE
Show method names in docs search bar

### ISSUE THREAD

**Issue by richardeakin** [cinder/Cinder#1244]
As the screencap shows below, when searching for a method name, classes will show up that contain a matching method, but the entries don't actually contain the method name. This is confusing to some users (#1243).

![screenshot 2015-12-21 20 25 56](https://cloud.githubusercontent.com/assets/5132127/11945770/55343180-a821-11e5-9c5b-6c29e4f0d7cd.png)

**Comment by drewish** [comment]
I'm happy to help with this if someone can point me in the right direction.

**Comment by gregkepler** [comment]
Yeah, I think we definitely need to think about how best to make search a bit more useful. This may require that we add more info to the generated search_index.js file. So instead of just saying that "mouseDown" is "tag" for the ArcBall page, we also add info that it's a function.

The tricky thing is that we don't want to increase the size of the search index file too much since that file gets parsed as soon as the page loads. We'll have to think about the best approach so that we can add more functionality like this as well as some other features that have been banging around in our heads.

@drewish If you want to familiarize yourself with how search works, may be it's something we can work on together. Happy to chat more about it in depth.

**Comment by gregkepler** [comment]
The best approach for updating search may be use server-side search like Solr. It can be more robust that way. It will require front-end javascript updates as well. Tests using solr are being done right now.
